src/coreclr/jit/codegenriscv64.cpp

```diff
@@ -4943,125 +4942,72 @@ void CodeGen::genPutArgStk(GenTreePutArgStk* treeNode)
-            regNumber addrReg = REG_NA;
+            regNumber loReg = treeNode->ExtractTempReg();
 
-            GenTreeLclVarCommon* varNode  = nullptr;
-            GenTree*             addrNode = nullptr;
+            GenTreeLclVarCommon* srcLclNode = nullptr;
+            regNumber            addrReg    = REG_NA;
+            ClassLayout*         layout     = nullptr;
 
-            if (source->OperGet() == GT_LCL_VAR)
+            // Setup "layout", "srcLclNode" and "addrReg".
+            if (source->OperIsLocalRead())
             {
-                varNode = source->AsLclVarCommon();
+                srcLclNode        = source->AsLclVarCommon();
+                layout            = srcLclNode->GetLayout(compiler);
+                LclVarDsc* varDsc = compiler->lvaGetDesc(srcLclNode);
+
+                // This struct must live on the stack frame.
+                assert(varDsc->lvOnFrame && !varDsc->lvRegister);
             }
             else // we must have a GT_BLK
             {
-                assert(source->OperGet() == GT_BLK);
-
-                addrNode = source->AsOp()->gtOp1;
-
-                // addrNode can either be a GT_LCL_ADDR<0> or an address expression
-                //
-                if (addrNode->IsLclVarAddr())
-                {
-                    // We have a GT_BLK(GT_LCL_ADDR<0>)
-                    //
-                    // We will treat this case the same as above
-                    // (i.e if we just had this GT_LCL_VAR directly as the source)
-                    // so update 'source' to point this GT_LCL_ADDR node
-                    // and continue to the codegen for the LCL_VAR node below
-                    //
-                    varNode  = addrNode->AsLclVarCommon();
-                    addrNode = nullptr;
-                }
-                else // addrNode is used
-                {
-                    // Generate code to load the address that we need into a register
-                    genConsumeAddress(addrNode);
-                    addrReg = addrNode->GetRegNum();
-                }
+                layout  = source->AsBlk()->GetLayout();
+                addrReg = genConsumeReg(source->AsBlk()->Addr());
             }
 
-            // Either varNode or addrNOde must have been setup above,
-            // the xor ensures that only one of the two is setup, not both
-            assert((varNode != nullptr) ^ (addrNode != nullptr));
+            unsigned srcSize = layout->GetSize();
 
-            ClassLayout* layout;
-
-            // unsigned gcPtrCount; // The count of GC pointers in the struct
-            unsigned srcSize;
-
-            // gcPtrCount = treeNode->gtNumSlots;
-            // Setup the srcSize and layout
-            if (source->OperGet() == GT_LCL_VAR)
+            // If we have an HFA we can't have any GC pointers,
+            // if not then the max size for the struct is 16 bytes
+            if (compiler->IsHfa(layout->GetClassHandle()))
```

Comment:
There are no HFAs on RISCV, so this checking can be deleted.

Reply:
Thank you!